src/app/ui/editor/moving_pixels_state.cpp

```diff
@@ -285,6 +285,13 @@ bool MovingPixelsState::onMouseDown(Editor* editor, MouseMessage* msg)
   Decorator* decorator = static_cast<Decorator*>(editor->decorator());
   Doc* document = editor->document();
 
+  // Return if it cannot draw in current layer
+  Layer* layer = editor->getSite().layer();
```

Comment:
Here using `Editor::layer()` will be better (same for other cases):
```suggestion
  Layer* layer = editor->layer();
```
I generally avoid creating the `Site` object (because it contains too much information, range of selected layers/frames from the timeline, etc.).